src/java/voldemort/server/protocol/admin/AsyncOperationService.java

```diff
@@ -81,21 +81,28 @@ public synchronized void submitOperation(int requestId, AsyncOperation operation
      * @param requestId Id of the request
      * @return True if request is complete, false otherwise
      */
-    public synchronized boolean isComplete(int requestId) {
-        if(!operations.containsKey(requestId))
-            throw new VoldemortException("No operation with id " + requestId + " found");
+    public synchronized boolean isComplete(int requestId, boolean remove) {
+        //if requestId not found, return true
+        if (!operations.containsKey(requestId))
```

Comment:
I take it back, the caller is not modified. The behavior needs to be preserved in the handleAsyncStatus

Felix, the problem is readonly fetch calls isComplete which has side effects. If fetch read only removes it pre-maturely, then handleAsyncStatus will throw.